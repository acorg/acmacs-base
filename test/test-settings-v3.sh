#! /bin/bash

TDIR=$(mktemp -d -t XXXXXX)
TESTDIR=$(dirname $0)

# ======================================================================

function on_exit
{
    if [[ -d "${TDIR}" ]]; then
        rm -rf "${TDIR}"
    fi
}

trap on_exit EXIT

function failed
{
    echo ERROR: "$@" FAILED >&2
    exit 1
}

trap failed ERR

# ======================================================================

cd "${TESTDIR}"

../dist/test-settings-v3 ./test-settings-v3.1.json main "$@" >"${TDIR}/test-settings-v3.output.txt"
if ! cmp "${TDIR}/test-settings-v3.output.txt" test-settings-v3.expected.txt; then
    diff "${TDIR}/test-settings-v3.output.txt" test-settings-v3.expected.txt || true
    echo
    echo
    ../dist/test-settings-v3
    false
fi
