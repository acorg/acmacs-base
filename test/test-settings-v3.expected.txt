{"shape": CIRCLE, "shown": true, "fill": "#00ffc0", "outline": "green", "outline_width": 1, "size": 5, "aspect": 1, "rotation": 0, "label": {"shown": true, "offset": {0, 1}, "size": 10, "color": black, "rotation": 0, "interline": 0.20, "style": {"slant": normal, "weight": normal, "family": }}, "label_text": ""}
{"clade": "$clade1"}
{"shape": CIRCLE, "shown": true, "fill": "#00ffc0", "outline": "red", "outline_width": 1, "size": 20, "aspect": 1, "rotation": 0, "label": {"shown": true, "offset": {0, 1}, "size": 10, "color": black, "rotation": 0, "interline": 0.20, "style": {"slant": normal, "weight": normal, "family": }}, "label_text": ""}
{"clade": "$clade1"}
{"shape": CIRCLE, "shown": true, "fill": "#7fffdf", "outline": "green", "outline_width": 1, "size": 5, "aspect": 1, "rotation": 0, "label": {"shown": true, "offset": {0, 1}, "size": 10, "color": black, "rotation": 0, "interline": 0.20, "style": {"slant": normal, "weight": normal, "family": }}, "label_text": ""}
{"clade": "$clade2"}
{"shape": CIRCLE, "shown": true, "fill": "magenta", "outline": "green", "outline_width": 1, "size": 5, "aspect": 1, "rotation": 0, "label": {"shown": true, "offset": {0, 1}, "size": 10, "color": black, "rotation": 0, "interline": 0.20, "style": {"slant": normal, "weight": normal, "family": }}, "label_text": ""}
{"clade": "$clade1"}
{"shape": CIRCLE, "shown": true, "fill": "cyan", "outline": "green", "outline_width": 1, "size": 5, "aspect": 1, "rotation": 0, "label": {"shown": true, "offset": {0, 1}, "size": 10, "color": black, "rotation": 0, "interline": 0.20, "style": {"slant": normal, "weight": normal, "family": }}, "label_text": ""}
{"clade": "$clade1"}
{"shape": CIRCLE, "shown": true, "fill": "cyan", "outline": "green", "outline_width": 1, "size": 5, "aspect": 1, "rotation": 0, "label": {"shown": true, "offset": {0, 1}, "size": 10, "color": black, "rotation": 0, "interline": 0.20, "style": {"slant": normal, "weight": normal, "family": }}, "label_text": ""}
{"clade": "$clade1"}
{"shape": CIRCLE, "shown": true, "fill": "yellow", "outline": "green", "outline_width": 1, "size": 5, "aspect": 1, "rotation": 0, "label": {"shown": true, "offset": {0, 1}, "size": 10, "color": black, "rotation": 0, "interline": 0.20, "style": {"slant": normal, "weight": normal, "family": }}, "label_text": ""}
{"clade": "$clade1"}
{"shape": CIRCLE, "shown": true, "fill": "#7fffdf", "outline": "green", "outline_width": 1, "size": 5, "aspect": 1, "rotation": 0, "label": {"shown": true, "offset": {0, 1}, "size": 10, "color": black, "rotation": 0, "interline": 0.20, "style": {"slant": normal, "weight": normal, "family": }}, "label_text": ""}
{"clade": "$clade1"}
